src/linter.cc

```diff
@@ -104,36 +109,69 @@ LinterResult ParseNoLintComments(absl::string_view sql,
   return result;
 }
 
+LinterResult CheckParserSucceeds(absl::string_view sql,
+                                 LinterOptions* options) {
+  ParseResumeLocation location = ParseResumeLocation::FromStringView(sql);
+  bool is_the_end = false;
+  int byte_position = -1;
+  LinterResult result;
+
+  while (!is_the_end) {
+    std::unique_ptr<ParserOutput> output;
+    byte_position = location.byte_position();
+    absl::Status status = ParseNextScriptStatement(&location, ParserOptions(),
+                                                   &output, &is_the_end);
+    if (!status.ok()) {
+      if (options->IsActive(ErrorCode::kParseFailed, byte_position)) {
+        ErrorLocation position = internal::GetPayload<ErrorLocation>(status);
+        result.Add(ErrorCode::kParseFailed, position.line(), position.column(),
+                   status.message());
+        return result;
+      }
+    }
+    options->AddParserOutput(output.release());
+  }
+
+  options->SetRememberParser(true);
+  return result;
+}
+
 LinterOptions GetOptionsFromConfig(Config config, absl::string_view filename) {
-  LinterOptions option(filename);
-  if (config.has_tab_size()) option.SetTabSize(config.tab_size());
+  LinterOptions options(filename);
+  if (config.has_tab_size()) options.SetTabSize(config.tab_size());
 
-  if (config.has_end_line()) option.SetLineDelimeter(config.end_line()[0]);
+  if (config.has_end_line()) options.SetLineDelimeter(config.end_line()[0]);
 
-  if (config.has_line_limit()) option.SetLineLimit(config.line_limit());
+  if (config.has_line_limit()) options.SetLineLimit(config.line_limit());
 
   if (config.has_allowed_indent())
-    option.SetAllowedIndent(config.allowed_indent()[0]);
+    options.SetAllowedIndent(config.allowed_indent()[0]);
 
-  if (config.has_single_quote()) option.SetSingleQuote(config.single_quote());
+  if (config.has_single_quote()) options.SetSingleQuote(config.single_quote());
 
   if (config.has_upper_keyword())
-    option.SetUpperKeyword(config.upper_keyword());
+    options.SetUpperKeyword(config.upper_keyword());
 
   std::map<std::string, ErrorCode> error_map = GetErrorMap();
 
   for (std::string check_name : config.nolint()) {
     if (error_map.count(check_name)) {
-      option.DisactivateCheck(error_map[check_name]);
+      options.DisactivateCheck(error_map[check_name]);
```

Comment:
nit: "disactivate" is almost never used in English. The used alternative is "deactivate", but in this context I'd say "DisableCheck"

Reply:
fixed